enhancers_pod.go

```diff
@@ -71,46 +38,225 @@ func runPodEnhancer(ctx context.Context, podMeta *v1.ObjectReference, cachedObje
 		}, breadcrumbLimit)
 	}
 
+	// Enhance message with pod name
 	message := sentryEvent.Message
-	sentryEvent.Message = fmt.Sprintf("%s: %s", podName, sentryEvent.Message)
+	sentryEvent.Message = fmt.Sprintf("%s: %s", pod.Name, sentryEvent.Message)
 
 	logger.Trace().Msgf("Current fingerprint: %v", sentryEvent.Fingerprint)
 
+	// Add node name as tag
+	nodeName := pod.Spec.NodeName
+	setTagIfNotEmpty(scope, "node_name", nodeName)
+
+	// Clean-up the object
+	pod.ManagedFields = []metav1.ManagedFieldsEntry{}
+	metadataJson, err := prettyJson(pod.ObjectMeta)
+	if err == nil {
+		scope.SetContext("Pod", sentry.Context{
+			"Metadata": metadataJson,
+		})
+	}
+
 	// Adjust fingerprint.
 	// If there's already a non-empty fingerprint set, we assume that it was set by
 	// another enhancer, so we don't touch it.
 	if len(sentryEvent.Fingerprint) == 0 {
 		sentryEvent.Fingerprint = []string{message}
 	}
 
-	// Using finger print to group events together
-	// The pod is not owned by a higher resource
-	if len(pod.OwnerReferences) == 0 {
-		// Standalone pod => most probably it has a unique name
-		sentryEvent.Fingerprint = append(sentryEvent.Fingerprint, podName)
-		// The pod is owned by a higher resource
-	} else {
-		// Check if pod is part of cronJob (as grandchild workload resource)
-
-		// check if a cronJob
-		var ok bool
-		if pod.OwnerReferences[0].Kind == "Job" {
-			ok, err = runCronsDataHandler(ctx, scope, pod, sentryEvent)
-			if err != nil {
-				return err
-			}
+	// Find the root owners and their kinds
+	rootOwners, err := findRootOwners(ctx, &KindObjectPair{
+		kind:   "Pod",
+		object: pod,
+	})
+	if err != nil {
+		return err
+	}
+
+	// Call specific enhancers for all root owners
+	// (there most likely is just one root owner)
+	for _, rootOwner := range rootOwners {
+		switch rootOwner.kind {
+		case "Pod":
+			sentryEvent.Fingerprint = append(sentryEvent.Fingerprint, rootOwner.object.GetName())
+		case "ReplicaSet":
+			ReplicaSetPodEnhancer(ctx, scope, rootOwner.object, sentryEvent)
+		case "Deployment":
+			deploymentPodEnhancer(ctx, scope, rootOwner.object, sentryEvent)
+		case "Job":
+			jobPodEnhancer(ctx, scope, rootOwner.object, sentryEvent)
+		case "CronJob":
+			cronjobPodEnhancer(ctx, scope, rootOwner.object, sentryEvent)
```

Comment:
Do we actually need to do the switch-case here? All those `*PodEnhancer` functions look quite similar, and probably can be generalized.

From a different angle: should we instead implement standalone enhancers for Deployments, Cronjobs, etc? The idea would be:
* If we're processing an event that involves a given Deployment, we'll run the Deployment enhancer on it.
* If we're processing a pod that belongs to a given Deployment (you already have the owner detection logic in place), we'll also run the same Deployment enhancer